stock_release_channel/models/stock_picking.py

```diff
@@ -103,27 +104,43 @@ def _find_release_channel_possible_candidate(self):
         self.ensure_one()
         return (
             self.env["stock.release.channel"]
-            .search(self._get_release_channel_possible_candidate_domain())
+            .search(self._release_channel_possible_candidate_domain)
             .sorted(key=lambda r: (not bool(r.partner_ids), r.sequence))
         )
 
-    def _get_release_channel_possible_candidate_domain_channel(self):
-        """Domain for finding channel candidates based on channel.
+    @property
+    def _release_channel_possible_candidate_domain(self):
+        """Domain for finding channel candidates based on picking
 
         The condition is defined by the channel.
         """
-        return [
+        self.ensure_one()
+        domain_base = self._release_channel_possible_candidate_domain_base
+        domain = [
             ("is_manual_assignment", "=", False),
-            ("state", "!=", "asleep"),
+            ("state", "in", ("open", "locked")),
```

Comment:
:heart: This change will fix potential unexpected behaviour when the addon `stock_release_channel_shipment_advice_deliver` is installed since it adds additional state :thinking:

Reply:
Indeed,  I can see it in the following changes